service/frontend/wrappers/clusterredirection/policy.go

```diff
@@ -78,8 +81,15 @@ const (
 type (
 	// ClusterRedirectionPolicy is a DC redirection policy interface
 	ClusterRedirectionPolicy interface {
-		WithDomainIDRedirect(ctx context.Context, domainID string, apiName string, requestedConsistencyLevel types.QueryConsistencyLevel, call func(string) error) error
-		WithDomainNameRedirect(ctx context.Context, domainName string, apiName string, requestedConsistencyLevel types.QueryConsistencyLevel, call func(string) error) error
+		Redirect(
+			ctx context.Context,
+			domainEntry *cache.DomainCacheEntry,
+			workflowExecution *types.WorkflowExecution,
```

Comment:
Nit: Based on my reading, domainEntry is mandatory but workflowExecution and actClSelPolicyForNewWF may both be nil. And they're mutually exclusive. It would be good to document these expectations